File: src/main/scala/li/cil/oc/common/item/GraphicsCard.scala
package li.cil.oc.common.item

import li.cil.oc.common.item.abstracts.SimpleItem
import net.minecraft.item.Item
import net.minecraft.item.Item.Properties
import net.minecraftforge.common.extensions.IForgeItem

class GraphicsCard(props: Properties, val tier: Int) extends SimpleItem(props) with IForgeItem with traits.ItemTier with traits.GPULike {
  @Deprecated
  override def getDescriptionId = super.getDescriptionId + tier

  override def gpuTier = tier

  override protected def tooltipName = Option(unlocalizedName)
}

File: src/main/scala/li/cil/oc/common/item/GraphicsCard.java
package li.cil.oc.common.item;

import li.cil.oc.common.item.abstracts.SimpleItem;
import li.cil.oc.common.item.interfaces.GPULike;
import li.cil.oc.common.item.interfaces.ItemTier;
import net.minecraftforge.common.extensions.IForgeItem;
import scala.Option;

import java.util.List;

public class GraphicsCard extends SimpleItem implements IForgeItem, ItemTier, GPULike {
    private final int tier;
    public GraphicsCard(Properties props, int tier) {
        super(props);
        this.tier = tier;
    }

    @Override
    public String getDescriptionId() {
        return super.getDescriptionId() + tier;
    }

    @Override
    public int gpuTier() {
        return tier;
    }

    @Override
    public Option<String> tooltipName() {
        return Option.apply(unlocalizedName);
    }

    @Override
    public List<Object> tooltipData() {
        return gpuTooltipData();
    }
}